### PR TITLE
Correct workflow badge 'event' example

### LINKED ISSUE #12329: Correct workflow badge 'event' example to use 'on'
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge

### What part(s) of the article would you like to see updated?

The example at https://github.com/github/docs/blob/27d1c2622074eba7ad5f74ddb474ea995b7da04c/content/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge.md?plain=1#L41-L47 is incorrect.

Using a URL such as <https://github.com/github/docs/actions/workflows/main.yml/badge.svg?event=pull_request> shows no status, as with this repo:

test-build-validate PR

Changing it to a URL such as <https://github.com/danielfdickinson/hugo-action-check-links/actions/workflows/test-check-links.yml/badge.svg?on=pull_request> shows the status of the event (in this case ``pull_request``)

test-build-validate PR

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by danielfdickinson**
### Why:

Closes https://github.com/github/docs/issues/12329

### What's being changed:

As described in issue #123129, Using a URL such as <https://github.com/github/docs/actions/workflows/main.yml/badge.svg?event=pull_request> shows no status, as with this badge: test-check-links PR.

Changing it to a URL such as <https://github.com/danielfdickinson/hugo-action-check-links/actions/workflows/test-check-links.yml/badge.svg?on=pull_request> shows the status of the event (in this case ``pull_request``) as shown with the following badge: test-check-links PR.

### Check off the following:

- [x] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Review by D-rock93 — COMMENTED** [review]
Help fix my so I can make $ plz

**Comment by ramyaparimi** [comment]
@danielfdickinson 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

File: content/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge.md [hunk 1]
```diff
@@ -38,10 +38,10 @@ This Markdown example adds a status badge for a branch with the name `feature-1`
 ![example branch parameter](https://github.com/github/docs/actions/workflows/main.yml/badge.svg?branch=feature-1)
 ```
 
-## Using the `event` parameter
+## Using the `on` parameter
```
Comment (ethomson) [inline content/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge.md:41]:
```suggestion
## Using the `event` parameter
```

File: content/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge.md [hunk 2]
```diff
@@ -38,10 +38,10 @@ This Markdown example adds a status badge for a branch with the name `feature-1`
...
 
 This Markdown example adds a badge that displays the status of workflow runs triggered by the `pull_request` event.
 
 ```markdown
-![example event parameter](https://github.com/github/docs/actions/workflows/main.yml/badge.svg?event=pull_request)
+![example on parameter](https://github.com/github/docs/actions/workflows/main.yml/badge.svg?on=pull_request)
```
Comment (ethomson) [inline content/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge.md:46]:
```suggestion
![example on parameter](https://github.com/github/docs/actions/workflows/main.yml/badge.svg?event=push)
```

File: content/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge.md [hunk 3]
```diff
@@ -38,10 +38,10 @@ This Markdown example adds a status badge for a branch with the name `feature-1`
 ![example branch parameter](https://github.com/github/docs/actions/workflows/main.yml/badge.svg?branch=feature-1)
 ```
 
-## Using the `event` parameter
+## Using the `on` parameter
 
 This Markdown example adds a badge that displays the status of workflow runs triggered by the `pull_request` event.
```
Comment (ethomson) [inline content/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge.md:43]:
```suggestion
This Markdown example adds a badge that displays the status of continuous integration builds in your default branch; those are builds that are triggered by the `push` event (when a commit is pushed or when a pull request is merged) into the repository's default branch.
```

**Review by ethomson — CHANGES_REQUESTED** [review]

**Comment by ethomson** [comment]
`event` is the correct parameter.  `on` is ignored, which means that:

`https://github.com/danielfdickinson/hugo-action-check-links/actions/workflows/test-check-links.yml/badge.svg?on=pull_request`: 


is the equivalent of no parameter at all:
`https://github.com/danielfdickinson/hugo-action-check-links/actions/workflows/test-check-links.yml/badge.svg`:


because we ignore unknown parameters; so it's also the equivalent of a parameter like `foo`:
`https://github.com/danielfdickinson/hugo-action-check-links/actions/workflows/test-check-links.yml/badge.svg?foo=pull_request`:


Now to the question of why we are not showing you status for `event=pull_request`: because we default to the default branch when none is specified.  And your default branch has never been the source of a pull request.  If you add an explicit branch, now you'll see the status for that branch:

`https://github.com/danielfdickinson/hugo-action-check-links/actions/workflows/test-check-links.yml/badge.svg?event=pull_request&branch=pr-upload-logs-on-failure`: 

(We do not have a way to show a badge for pull requests _targetting_ a branch, eg `target=main`.  If this would be useful to you [I'd love to understand more about it](https://github.com/github/feedback/discussions/categories/actions-and-packages-feedback).

But I think that what most people want for the status badge here is _not_ based on the `pull_request` trigger at all, but on the `push` trigger.  (In other words, _is my main branch CI green?)  And our documentation should reflect _that_.

**Comment by danielfdickinson** [comment]
First, thank you for clarifying the on vs. event situation.

>But I think that what most people want for the status badge here is not based on the pull_request trigger at all, but on the push trigger. (In other words, _is my main branch CI green?) And our documentation should reflect that.

I guess I'm a little confused about how that differs from the default (no event) for a branch.

What I was hoping for, really, was that I could look at the badge in the README for a PR (for example if the README is deployed as part of the preview version of update docs) and see if the PR is CI green.

It's kind of corner case though, and I'll probably just drop using the event parameter in my own repos.

**Comment by danielfdickinson** [comment]
I've add https://github.com/github/feedback/discussions/9091 for what I was actually looking for (as suggested by @ethomson). I'm not quite sure what to do with the PR since the changes needed are completely different than proposed (basically needs to be rewritten as a clarification of the current docs, with the information provided by @ethomson). That probably should be done a new/separate PR?

**Comment by hubwriter** [comment]
> That probably should be done a new/separate PR?

Yes, I agree that it would be best to close this PR and raise a separate issue & PR for clarification of this area.

**Pull request closed without merging**